### DISCUSSION TITLE
way to combine TextLineEntry with TextBox ?

### DISCUSSION THREAD

**Discussion by Minedigger76** [MyreMylar/pygame_gui#277] [Q&A] [answered]
First of all, thanks a lot already for all work done. I'm not very experienced with coding, so I might ask the obvious: I'm looking for an interactive textmodule which combines the features of textlineentry with textboxlayout. I'm not able to get the changed text ( via delete, insert, .. ) from the textbox ? Is there any function which I might have missed ?

Thanks a lot for the info

**Comment by Minedigger76** [comment]
Dear, found the text embedded in a TextLineChunkFTFont object, but I cannot access the text ? How do I get access to this, I cannot find a method ? Many thanks for the help / explanation

**Comment by MyreMylar** [comment, marked as answer]
Hello,

At the minute there is no 'EditableTextBox' UIElement (I'm imagining something like the main window pane of windows notepad here). It is on the to do list for the future.

If you are looking for something like a text console (think a CMD/Powershell/command line) then there is an example of that in the examples project here:

https://github.com/MyreMylar/pygame_gui_examples

look at `text_console.py`

If you want to print all the lines of text you enter into the console window you could do something like:

```
def print_all_logged_lines(console):
    for line in console.logged_commands_above:
        print(line)
    if console.current_logged_command is not None:
        print(console.current_logged_command)
    for line in console.logged_commands_below:
        print(line)
```

and passing in the console window.

**Reply by Minedigger76** [reply]
Dear Dan,

Thanks a lot for the fast reply, the text_console option can do the job for now: it's not a showstopper. Great to read an 'EditableTextBox' UIElement is on the to-do list. I'll recheck the pygame_gui from time to time as it would improve the user experience.

Many thanks already for your time and energy.

KR,

Johan
